src/app/cluster/details/external-cluster/update-external-cluster-machine-deployment-dialog/component.ts

```diff
@@ -49,6 +55,8 @@ export class UpdateExternalClusterMachineDeploymentDialogComponent extends BaseF
   readonly Controls = Controls;
   disableReplicaControl: boolean;
   versions: string[] = [];
+  versionsLoaded = false;
+  private _state = KubeletVersionState.Loading;
```

Comment:
There's no need for this field to be private. This can be made public and `label` getter can be removed.
```suggestion
  versionsLabel = KubeletVersionState.Loading;
```